next/components/forms/segments/AccountNavBar/AccountNavBar.tsx

```diff
@@ -118,14 +228,182 @@ export const AccountNavBar = ({ className, currentLanguage }: IProps) => {
+      </div>
+      <ChevronDownSmall
+        className={`ml-3 hidden mix-blend-normal lg:flex ${
+          isSelectClicked && isComponentVisible && 'mb-1 -rotate-180'
+        }`}
+      />
+      {isSelectClicked && isComponentVisible && (
+        <div className="absolute top-12 -left-3 z-20 mt-1 flex h-auto cursor-default flex-col items-center justify-center lg:left-0">
+          <div className="flex h-auto min-h-[60px] w-full flex-col rounded-lg bg-white py-2 shadow-[0_8px_16px_rgba(0,0,0,0.12)]">
+            {menuItems.map((option) => (
+              <div key={option.id} className="cursor-pointer flex py-2 px-5">
+                <div className="flex relative flex-row items-start gap-2 rounded-xl p-4 bg-gray-50">
+                  <div className="flex h-2 w-2 items-center justify-center">
+                    <span>{option.icon}</span>
+                  </div>
+                </div>
+                <div
+                  className="text-p2 hover:text-p2-semibold text-font p-2 whitespace-nowrap"
+                  onClick={() => handleChange(option.id)}
+                >
+                  {t(option.title)}
+                </div>
+              </div>
+            ))}
+          </div>
+        </div>
+      )}
+    </div>
+  )
+}
+
+const LanguageSelect = ({
+  languages: options,
+  currentLanguage: current,
+  onLanguageChange: onChange,
+}: LanguageSelectProps) => {
+  const [isSelectClicked, setIsSelectClicked] = useState(false)
+  const { ref, isComponentVisible } = useComponentVisible(false, setIsSelectClicked)
+  const dropDownOptions = options?.filter((option) => option.key != current)
+  const handleChange: React.MouseEventHandler<HTMLDivElement> = (e) => {
+    if (!onChange) return
+
+    const selectedKey = e.currentTarget.innerText.toLowerCase()
+    const selectedOption = options?.find((opt) => opt.key === selectedKey)
+
+    if (selectedOption) {
+      onChange(selectedOption)
+      setIsSelectClicked(false)
+    }
+  }
+
+  if (!options) return null
+
+  const handleClick: React.MouseEventHandler<HTMLDivElement> = (e) => {
+    setIsSelectClicked(!isSelectClicked)
+  }
+
+  return (
+    <div className="relative flex w-12 cursor-pointer items-center" ref={ref} onClick={handleClick}>
+      <div className="font-light lg:font-semibold">{current?.toUpperCase()} </div>
+      <ChevronDownSmall
+        className={`ml-3 hidden mix-blend-normal lg:flex ${
+          isSelectClicked && isComponentVisible && 'mb-1 -rotate-180'
+        }`}
+      />
+      {isSelectClicked && isComponentVisible && (
+        <div className="absolute top-6 -left-3 z-20 mt-1 flex h-auto w-11 cursor-default flex-col items-center justify-center lg:left-0">
+          <div className="z-10 h-0 w-4 border-x-8 border-b-4 border-solid border-transparent border-b-[#F8D7D4]" />
+          <div className="flex h-auto min-h-[60px] w-full flex-col items-center rounded-lg bg-[#F8D7D4] pt-1 pb-3 shadow-[0_8px_24px_rgba(0,0,0,0.16)]">
```

Comment:
bg-[customcolor] should be replaced. I also think we should have all the required boxShadow configsin tailwind.config - you should be able to use one of them, or nudge the design if this one is not one of the preconfigured ones to go along with the ones in design system. Same elsewhere.